packages/tools/addon/components/cs-version-control.js

```diff
@@ -158,6 +159,28 @@ export default Component.extend({
     yield model.save();
   }).keepLatest(),
 
+  cancel: task(function * () {
+    this.get('tools').setEditing(false);
+
+    if (this.get('anythingDirty')) {
+      let model = this.get('model');
+
+      model.rollbackAttributes();
+      model.rollbackRelationships();
+
+      if (model.dirtyTrackingRelationNames) {
+        let relationNames = Object.keys(model.dirtyTrackingRelationNames);
+        relationNames.forEach(relation => {
+          let relationItems = model.get(relation);
+          relationItems.forEach(item => {
+            item.rollbackAttributes();
+            item.rollbackRelationships();
+          });
+        });
+      }
+    }
+  }),
```

Comment:
This version works for reseting all types of dirty fields except the titles editor on dotbc. Need help with writing tests, though. Getting `Uncaught TypeError: model.rollbackAttributes is not a function` error when I try to use the rollback functions in `cs-version-control-test` file.